server/src/main/java/org/elasticsearch/action/bulk/TransportSimulateBulkAction.java

```diff
@@ -360,6 +372,23 @@ private void validateUpdatedMappings(
                 0
             );
         });
+        final Collection<String> ignoredFields;
+        if (result == null) {
+            ignoredFields = List.of();
+        } else {
+            List<LuceneDocument> luceneDocuments = result.parsedDoc().docs();
+            if (luceneDocuments != null && luceneDocuments.size() == 1) {
```

Comment:
Should we add an `assert luceneDocuments().size() == 1` somewhere here to ensure that we fail if in the future a single index request results in more than one doc? (We'd silently ignored the response if we didn't)

Reply:
Sounds good.